db/hybrid_logical_vector.go

```diff
@@ -272,7 +335,13 @@ func (hlv *HybridLogicalVector) AddNewerVersions(otherVector HybridLogicalVector
 		// Iterate through incoming vector previous versions, update with the version from other vector
 		// for source if the local version for that source is lower
 		for i, v := range otherVector.PreviousVersions {
-			if hlv.PreviousVersions[i] == 0 || hlv.PreviousVersions[i] < v {
+			if hlv.PreviousVersions[i] == "" {
+				hlv.setPreviousVersion(i, v)
+			}
```

Comment:
Think we probably want to do this here?

```suggestion
			} else {
```

Reply:
Good spot!